### PR TITLE
feat: instrument with prometheus

### PR DESCRIPTION
**Pull request by id**

### CODE REVIEW

**Review by qzhuyan — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -161,6 +171,13 @@ Usage: emqtt_bench sub [--help <help>] [-d <dist>] [-h [<host>]] [-p [<port>]] [
                              0]
   --log_to                   Control where the log output goes. console: directly to the console      null: quietly, don't 
                              output any logs. [default: console]
+  --prometheus               Enable metrics collection via Prometheus.
+                             Usually used with --restapi to enable
+                             scraping endpoint. [default: false]
+  --restapi                  Enable REST API for monitoring and control.
```
Comment (zmstone) [inline README.md:177]:
```suggestion
  --restapi                  "host:port" ?
```

File: README.md [hunk 2]
```diff
@@ -98,7 +100,14 @@ Usage: emqtt_bench conn [--help <help>] [-d <dist>] [-h [<host>]]
...
                              don't output any logs. [default: console]
-
+  --prometheus               Enable metrics collection via Prometheus.
+                             Usually used with --restapi to enable
+                             scraping endpoint.
+  --restapi                  Enable REST API for monitoring and control. 
+                             For now only serves /metrics. 
+                             Can be set to IP:Port to listen on a specific IP and Port,
+                             or just Port to listen on all interfaces on
+                             that port. [default: disabled]
```
Comment (thalesmg) [inline README.md:110]:
```suggestion
  --restapi                  Enable REST API for monitoring and control. 
                             Only takes effect if Prometheus is enabled.
                             For now only serves /metrics. 
                             Can be set to IP:Port to listen on a specific IP and Port,
                             or just Port to listen on all interfaces on
                             that port. [default: disabled]
```
?

File: src/emqtt_bench.erl [hunk 3]
```diff
@@ -1231,22 +1270,71 @@ counters() ->
...
+               , pub_succ
+               , connect_succ
+               , connect_fail
+               , connect_retried
+               , reconnect_succ
+               , unreachable
+               , connection_refused
+               , connection_timeout
+               , connection_idle
+               ],
```
Comment (thalesmg) [inline src/emqtt_bench.erl:1300]: Nit: maybe we could move this list of counter names to a `counter_name/0` function, and use it both here and in `counters/0`?

**Review by thalesmg — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (id) [reply README.md:110]: > Only takes effect if Prometheus is enabled.

I was thinking to add other functionality later, like change config in runtime without restart. Could be useful if you already have 1M connections and don't want to redo them again.

File: src/emqtt_bench.erl (hunk 3 above)
Comment (id) [reply src/emqtt_bench.erl:1300]: Will do in the next PR. There are a few more things to deduplicate.

**Pull request merged**